plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -558,6 +567,96 @@ public void updatePartitionStatistics(Table table, String partitionName, Functio
         removedStatistics.forEach(column -> deletePartitionColumnStatistics(table.getDbName(), table.getTableName(), partitionName, column));
     }
 
+    @Override
+    public void updatePartitionStatistics(Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        if (updatePartitionStatisticsBatchEnabled) {
```

Comment:
You could make use of `com.google.common.collect.Lists#partition` here to massively simplify the code.

Here is a usage example:

https://github.com/trinodb/trino/blob/6310e5e1415fe0bc89444016fc516640c3a3fcbb/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java#L879-L886